File: assets/src/app/models/person.ts

export enum LinkLabel {
  facebook = "Facebook",
  imdb = "IMDb",
  linkedin = "LinkedIn",
}

export class Links {
  imdb?: string;
  facebook?: string;
  linkedin?: string;
}

export class Person {
  id: number;
  last_name: string;
  first_names: string[];
  gender: string;
  birth_date: string;
  birth_city: string;
  birth_country: string;
  nationalities: string[];
  links: Links;

  constructor() {
    this.gender = "Unknown";
    this.first_names = new Array<string>("");
    this.nationalities = new Array<string>();
    this.links = new Links;
  }
}

export class IMDbPeople {
  name: string;
  birth_date: string;
  birth_location: string;
  picture_url: string;
}

File: lib/ex_backend/persons/person.ex
defmodule ExBackend.Persons.Person do
  use Ecto.Schema
  import Ecto.Changeset
  alias ExBackend.Persons.Person

  schema "persons" do
    field(:last_name, :string)
    field(:first_names, {:array, :string})
    field(:gender, :string, default: "")
    field(:birth_date, :date)
    field(:birth_city, :string, default: "")
    field(:birth_country, :string, default: "")
    field(:nationalities, {:array, :string}, default: [])
    field(:links, :map, defaut: %{})

    timestamps()
  end

  @doc false
  def changeset(%Person{} = job, attrs) do
    job
    |> cast(attrs, [
      :last_name,
      :first_names,
      :gender,
      :birth_date,
      :birth_city,
      :birth_country,
      :nationalities,
      :links
    ])
    |> validate_required([:last_name, :first_names, :gender, :birth_date])
  end
end
